ts/features/itwallet/discovery/screens/ItwDiscoveryInfoScreen.tsx

```diff
@@ -28,26 +26,26 @@ export const ItwDiscoveryInfoScreen = ({
 }: ItwDiscoveryInfoScreenProps) => {
   const { isL3 = false } = route.params ?? {};
 
-  const machineRef = ItwEidIssuanceMachineContext.useActorRef();
-
   useFocusEffect(
     useCallback(() => {
       trackItWalletIntroScreen();
     }, [])
   );
 
-  useOnFirstRender(() => {
-    machineRef.send({ type: "start", isL3 });
-  });
```

Comment:
We moved the start event inside `ItwL3DiscoveryInfoComponent` and `ItwDiscoveryInfoComponent` components. Do we have advantages? I think we are duplicating code. What do you think?

Reply:
Right, I was missing the last part!
But can't this be solved by inverting the logic? Also I don't think this logic is correct, we are still tracking `trackItWalletIntroScreen` even if the user does not have NFC.

Reply:
> But can't this be solved by inverting NFC and L3 checks? I don't really like the idea of non-screen components sending events to the machine on mount.

If I reverse the order of the checks (checking for NFC before L3) I’m not able to support scenario 2B. Right? Anyway I totally agree with you, I’d prefer to avoid having non-screen components trigger events on mount.

> Also we are still tracking trackItWalletIntroScreen even if the user does not have NFC, which is not correct.

Good catch

Reply:
What if we put the content of `ItwL3DiscoveryInfoComponent` in  `ItwDiscoveryInfoScreen` and move the logic in `ItwDiscoveryInfoScreen`? Something like the following:

```js
useOnFirstRender(() => {
  if (!isL3) {
    machineRef.send({ type: "start", isL3: false });
  }

  if (hasNfcFeature) {
    machineRef.send({ type: "start", isL3: true });
  }
});
```

Could this work?

Reply:
Applied in https://github.com/pagopa/io-app/pull/6981/commits/1effc2a83d01c0ec2ca34185b9e81637e8221691, What do you think?